packages/starlight/utils/head.ts

```diff
@@ -25,6 +25,8 @@ function hasTag(head: HeadConfig, entry: HeadConfig[number]): boolean {
 			return head.some(({ tag }) => tag === 'title');
 		case 'meta':
 			return hasOneOf(head, entry, ['name', 'property', 'http-equiv']);
+		case 'link':
+			return head.some(({ attrs }) => typeof attrs.rel === 'string' && ['canonical'].includes(attrs.rel))
```

Comment:
Can’t this be simplified to this?

```suggestion
			return head.some(({ attrs }) => attrs.rel === 'canonical')
```

Reply:
I've applied the suggestions in this commit. https://github.com/withastro/starlight/pull/2154/commits/9d18b01ad2cf07c72177cd63f8314b423f44fc88